### DISCUSSION TITLE
TimeFrames? how long does it take for a submission to reach winget?

### DISCUSSION THREAD

**Discussion by xxPYROxxJONESxx** [microsoft/winget-pkgs#19502] [Q&A] [answered]
Hi, very new to all this. had a good search and couldn't find an answer...

So my question is, what sort of timeframes does Winget-Create work on? so I've submitted a package and have read that these are undergoing a final human review these days but how long do these things take? just curious. I had a look through the pull requests for programs like MSI Afterburner and EVGA PrecisionX1 and if I remember correctly they both seemed to be approved... yet when I search for them in Winget, they're nowhere to be seen... if anyone could clear that up for me I would appreciate it greatly.

**Comment by ImJoakim** [comment, marked as answer]
> So my question is, what sort of timeframes does Winget-Create work on? so I've submitted a package and have read that these are undergoing a final human review these days but how long do these things take? just curious. 

The amount of time it will take for the whole process after submitting a PR till it's merged various. It starts by going through 7 automated checks. This specific process usually takes 30-40 minutes if the manifest is successful. Your specific PR (#19499) took 34 minutes. The human review also various depending on when the Administrators and Community Moderators are available and going through them - see #15674 for more information regarding moderation. From my experience, this only takes a few hours. I have not encountered it taking more than a day. Some times, it takes just a few minutes after it passed the 7 checks. But again, as this is a human review, it various.

> I had a look through the pull requests for programs like MSI Afterburner and EVGA PrecisionX1 and if I remember correctly they both seemed to be approved... yet when I search for them in Winget, they're nowhere to be seen... if anyone could clear that up for me I would appreciate it greatly.

The MSI Afterburner was removed as the download was from an unofficial link - see #15702.
The EVGA PrecisionX1 was removed because the installation failed - see #19263.

**Reply by xxPYROxxJONESxx** [reply]
thank you so much for your prompt reply, good sir. So just to be clear, after receiving human approval they're available via Winget almost immediately? or are programs added in bulk on some kind of schedule? 

ah, I must have misread their current status when I was looking at those programs - I thought they were in the clear.

**Reply by ImJoakim** [reply]
> thank you so much for your prompt reply, good sir. So just to be clear, after receiving human approval they're available via Winget almost immediately? or are programs added in bulk on some kind of schedule?

They will be merged automatically after being approved. Example from one of my PRs (#19405):
Here you can see the PR being approved by OfficialEsco and merged automatically by the bot, making it available through winget.

![image](https://user-images.githubusercontent.com/52826317/124194260-871bea00-dac8-11eb-8393-6937ef620366.png)

**Reply by denelon** [reply]
The "publishing pipeline" takes all of the recently merged manifests and packages them up for publishing. In most cases it takes slightly less than an hour. As soon as publishing is complete, we usually have new manifests so the pipeline runs again. This cycle continues until there are no new merged manifests to publish. 

We're working on building some badges to help show progress and give a little more visibility into the health and activities of the various services supporting the Windows Package Manager.
